### DISCUSSION TITLE
Download file from memory (eg. bytesio) via ui.download

### DISCUSSION THREAD

**Discussion by italanchan** [zauberzeug/nicegui#1275] [Ideas / Feature Requests]
Is ui.download support bytesio or bytes array as file source?

**Comment by rodja** [comment]
Currently `ui.download` only supports urls. However, we have an example on how to use it with bytesio: https://github.com/zauberzeug/nicegui/blob/main/examples/download_text_as_file/main.py

There is also a (stale) PR to allow downloading local files: https://github.com/zauberzeug/nicegui/pull/1118. It is a bit tricky, because `ui.download` is not a real element but more like a function...

**Comment by rodja** [comment]
We have found a solution to download local files #1118. I'll convert this into a feature request. Anyone interested in implementing download `bytesio` on top of #1118?

**Comment by italanchan** [comment]
if this support bytesio, no more temp files saved and remove lately.

**Comment by falkoschindler** [comment]
This feature has been implemented in PR #2050.
